lib/src/test/kotlin/starknet/utils/LegacyDevnetClient.kt

```diff
@@ -0,0 +1,360 @@
+package starknet.utils
+
+import com.swmansion.starknet.data.types.Felt
+import com.swmansion.starknet.data.types.GetBlockHashAndNumberResponse
+import com.swmansion.starknet.data.types.transactions.*
+import com.swmansion.starknet.service.http.HttpService
+import com.swmansion.starknet.service.http.OkHttpService
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.*
+import starknet.utils.data.*
+import starknet.utils.data.serializers.AccountDetailsSerializer
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.*
+import java.util.concurrent.TimeUnit
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.exists
+import kotlin.io.path.readText
+
+class LegacyDevnetClient(
+    private val host: String = "0.0.0.0",
+    private val port: Int = 5050,
+    private val httpService: HttpService = OkHttpService(),
+    private val accountDirectory: Path = Paths.get("src/test/resources/account"),
+) : AutoCloseable {
+    private val baseUrl: String = "http://$host:$port"
+    private val seed: Int = 1053545547
+    private val json = Json { ignoreUnknownKeys = true }
+
+    private lateinit var devnetProcess: Process
+
+    private var isDevnetRunning = false
+
+    lateinit var accountDetails: AccountDetails
+
+    val gatewayUrl: String = "$baseUrl/gateway"
+    val feederGatewayUrl: String = "$baseUrl/feeder_gateway"
+    val rpcUrl: String = "$baseUrl/rpc"
+
+    @Serializable
+    data class Block(
+        @SerialName("block_hash") val hash: Felt,
+        @SerialName("block_number") val number: Int,
+    )
+
+    data class TransactionResult(
+        val address: Felt,
+        val transactionHash: Felt,
+    )
+
+    fun start() {
+        if (isDevnetRunning) {
+            throw DevnetSetupFailedException("Devnet is already running")
+        }
+
+        // This kills any zombie devnet processes left over from previous
+        // test runs, if any.
+        ProcessBuilder(
+            "pkill",
+            "-f",
+            "starknet-devnet.*$port.*$seed",
+        ).start().waitFor()
+
+        devnetProcess =
+            ProcessBuilder(
+                "starknet-devnet",
+                "--disable-rpc-request-validation",
+                "--host",
+                host,
+                "--port",
+                port.toString(),
+                "--seed",
+                seed.toString(),
+                "--sierra-compiler-path",
+                "src/test/resources/compilers/v2/cairo/bin/starknet-sierra-compile",
```

Comment:
do we need legacy devnet to test cairo1/2 contracts? shouldn't it be doable from devnet-rs only?

Reply:
We need legacy devnet for all of the gateway tests (until it's finally removed), that includes declaring and estimating fees for Cairo 1/2.

Reply:
sure, makes sense. I was just under the impression that we'll just not test gateway anymore, since it will be removed in a month. If you feel safer having those though, I won't oppose

Reply:
That was the initial idea, but since then the date of gateway removal got pushed back few times, I think we should act on it once we know for sure.